File: archive/legacy-review/gmx-waysun/eeg-waysun-leaderboard/eeg-waysun-leaderboard/common/src/main/scala/stella/leaderboard/models/AggregationResult.scala
package stella.leaderboard.models

import java.time.OffsetDateTime

import spray.json.DefaultJsonProtocol._
import spray.json.DefaultJsonProtocol.jsonFormat11
import spray.json.RootJsonFormat
import sttp.tapir.Schema

import stella.common.http.json.JsonFormats.offsetDateTimeFormat

final case class AggregationResult(
    position: Int,
    groupByFieldValue: String,
    windowRangeStart: Option[OffsetDateTime],
    windowRangeEnd: Option[OffsetDateTime],
    min: Float,
    max: Float,
    count: Int,
    sum: Float,
    custom: String,
    createdAt: OffsetDateTime,
    updatedAt: OffsetDateTime)

object AggregationResult {

  implicit lazy val aggregationResultFormat: RootJsonFormat[AggregationResult] = jsonFormat11(AggregationResult.apply)

  implicit lazy val aggregationResultSchema: Schema[AggregationResult] = Schema.derived[AggregationResult]
}

File: archive/legacy-review/gmx-waysun/eeg-waysun-stella-frontend/eeg-waysun-stella-frontend/apps/mock-server/src/mocked_data/aggregationResult.ts
import { AggregationResultType } from "utils";

export const aggregationResultPayload: AggregationResultType = {
  "pageNumber": 1,
  "numberOfPages": 3,
  "pageSize": 3,
  "results": [
    {
      "position": 1,
      "groupByFieldValue": "Ronaldo123",
      "min": 12,
      "max": 100,
      "count": 360,
      "sum": 5436.09,
      "custom": "",
      "createdAt": "2021-01-23T01:23:45.678+09:00",
      "updatedAt": "2021-01-23T01:34:40.112+09:00"
    },
    {
      "position": 2,
      "groupByFieldValue": "Stefan_111",
      "min": 1,
      "max": 99,
      "count": 360,
      "sum": 1236.1,
      "custom": "",
      "createdAt": "2021-01-23T01:23:45.678+09:00",
      "updatedAt": "2021-01-23T01:34:40.112+09:00"
    },
    {
      "position": 3,
      "groupByFieldValue": "Feeboo_N4c1",
      "min": 77,
      "max": 97,
      "count": 360,
      "sum": 936.54,
      "custom": "",
      "createdAt": "2021-01-23T01:23:45.678+09:00",
      "updatedAt": "2021-01-23T01:34:40.112+09:00"
    },
  ]
}